python/ray/workflow/api.py

```diff
@@ -586,10 +586,33 @@ def create(dag_node: "DAGNode", *args, **kwargs) -> Workflow:
     """
     from ray.workflow.dag_to_workflow import transform_ray_dag_to_workflow
 
+    if not isinstance(dag_node, DAGNode):
+        raise TypeError("Input should be a DAG.")
     input_context = DAGInputData(*args, **kwargs)
     return transform_ray_dag_to_workflow(dag_node, input_context)
 
 
+@PublicAPI(stability="beta")
+def continuation(dag_node: "DAGNode") -> Union[Workflow, ray.ObjectRef]:
+    """Converts a DAG into a continuation.
+
+    The result depends on the context. If it is inside a workflow, it
+    returns a workflow; otherwise it executes and get the result of
```

Comment:
Sorry, I just got a chance to look at this PR. I have one question about the semantics of continuation.

Maybe I miss something, but shouldn't the inner one return object ref instead of dereferencing it? Otherwise, the semantics seems not aligned with ray.

Reply:
I think the semantics of `continuation` is to dereferencing it instead of returning an object ref. That's the semantics we want for DAG, and it is aligned both for ray and workflow.